### PR TITLE
[backport-111] SCA hardening for modular field inversion in EC_GROUPs

### PR DESCRIPTION
**Pull request by romen**
Backport of #8254 to `1.1.1-stable`.

Cherry-picking required manual changes in:
- `CHANGES`
-  `test/build.info`

### CODE REVIEW

File: crypto/ec/ec_lcl.h [hunk 1]
```diff
@@ -154,6 +154,13 @@ struct ec_method_st {
     int (*field_sqr) (const EC_GROUP *, BIGNUM *r, const BIGNUM *a, BN_CTX *);
     int (*field_div) (const EC_GROUP *, BIGNUM *r, const BIGNUM *a,
                       const BIGNUM *b, BN_CTX *);
+    /*-
+     * 'field_inv' computes the multipicative inverse of a in the field,
+     * storing the result in r.
+     *
+     * If 'a' is zero (or equivalent), you'll get an EC_R_CANNOT_INVERT error.
+     */
+    int (*field_inv) (const EC_GROUP *, BIGNUM *r, const BIGNUM *a, BN_CTX *);
```
Comment (bernd-edlinger) [inline crypto/ec/ec_lcl.h]: you might want to change the copyright year ?

Comment (romen) [reply crypto/ec/ec_lcl.h]: Good point!

**Comment by romen** [comment]
@mattcaswell I pushed a fixup commit with updated copyright years as suggested by @bernd-edlinger , could you reapprove?

**Comment by romen** [comment]
Merged to `1.1.1-stable`:
- 48e82c8e22f8ac16ad0666d99172f6dbaf33953a
- d11e4bcddd3b235e0ca87eb0251a1e5136d95c70

Thanks everyone!

**Pull request closed without merging**